graph/schema.resolvers.go

```diff
@@ -0,0 +1,282 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/Matterhorn-Apps/MatterhornApiService/database"
+	"github.com/Matterhorn-Apps/MatterhornApiService/graph/generated"
+	"github.com/Matterhorn-Apps/MatterhornApiService/graph/model"
+)
+
+func (r *mutationResolver) CreateUser(ctx context.Context, input model.NewUser) (*model.User, error) {
+	// Query the database for matching exercise records
+	query := fmt.Sprintf("INSERT INTO users (user_id) VALUES (%s);", input.ID)
+	_, readErr := r.DB.Exec(query)
+	if readErr != nil {
+		if errCode, ok := database.TryExtractMySQLErrorCode(readErr); ok {
+			switch *errCode {
+			case 1062:
+				// User ID exists
+				return nil, readErr
+			}
+		}
+
+		log.Printf("Failed to query database: %v", readErr)
+		return nil, readErr
+	}
+
+	return &model.User{
+		ID: input.ID,
+	}, nil
+}
+
+func (r *mutationResolver) CreateExerciseRecord(ctx context.Context, input *model.NewExerciseRecord) (*model.ExerciseRecord, error) {
+	if *input.Calories < 0 {
+		return nil, errors.New("Invalid calorie value provided")
+	}
+
+	// Query the database for matching exercise records
+	query := fmt.Sprintf("INSERT INTO exercise_records (user_id, label, calories) VALUES ('%s', '%s', %d);",
+		input.UserID, *input.Label, *input.Calories)
+	_, readErr := r.DB.Exec(query)
+	if readErr != nil {
+		if errCode, ok := database.TryExtractMySQLErrorCode(readErr); ok {
+			switch *errCode {
+			case 1452:
+				// User not found
+				return nil, readErr
+			}
+		}
+
+		log.Printf("Failed to query database: %v", readErr)
+		return nil, readErr
+	}
+
+	return &model.ExerciseRecord{
+		Calories: *input.Calories,
+		Label:    *input.Label,
+	}, nil
+}
+
+func (r *mutationResolver) CreateFoodRecord(ctx context.Context, input *model.NewFoodRecord) (*model.FoodRecord, error) {
+	if *input.Calories < 0 {
+		return nil, errors.New("Invalid calorie value provided")
+	}
+
+	// Query the database for matching Food records
+	query := fmt.Sprintf("INSERT INTO food_records (user_id, label, calories) VALUES ('%s', '%s', %d);",
+		input.UserID, *input.Label, *input.Calories)
+	_, readErr := r.DB.Exec(query)
+	if readErr != nil {
+		if errCode, ok := database.TryExtractMySQLErrorCode(readErr); ok {
+			switch *errCode {
+			case 1452:
```

Comment:
what is this integral case?

Reply:
This is detecting specific error codes from MySQL. Previously this was used to determine status codes for REST responses. Now it doesn't really serve any particular purpose, but I'm inclined to keep it around for now - we can perhaps add more specific and helpful log or error messages in the future for these expected cases.